utils/autotailor.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python3
+
+# Copyright (C) 2020 Matěj Týč <matyc@redhat.com>
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+# Boston, MA 02110-1301 USA
+
+import argparse
+import datetime
+import re
+import xml.etree.ElementTree as ET
+
+
+ALL_NS = {
+    "xccdf-1.1": "http://checklists.nist.gov/xccdf/1.1",
+    "xccdf-1.2": "http://checklists.nist.gov/xccdf/1.2",
+}
+DEFAULT_PROFILE_SUFFIX = "_customized"
+
+
+def id_is_full(string):
+    return re.match(r"^xccdf_org\.", string) is not None
+
+
+class Tailoring:
+    PROFILES_PREFIX = "xccdf_org.ssgproject.content_profile_"
+    RULES_PREFIX = "xccdf_org.ssgproject.content_rule_"
+    VARS_PREFIX = "xccdf_org.ssgproject.content_value_"
```

Comment:
These prefixes are specific for scap-security-guide. OpenSCAP aims to be a generic SCAP processing tool. I think it would be more beneficial if this is customizable. Then people could use it on any SCAP content.

Reply:
People can use it with any SCAP content if they provide valid XCCDF IDs as inputs. If invalid IDs are supplied, then this gets prepended.

Reply:
I have an idea. I think you can create an option `--reverse-dns` that would allow users to provide custom reverse DNSs.